src/Illuminate/Collections/Collection.php

```diff
@@ -1050,6 +1052,37 @@ public function splitIn($numberOfGroups)
         return $this->chunk(ceil($this->count() / $numberOfGroups));
     }
 
+    /**
+     * Get the first item in the collection, but only if exactly
+     * item exists. Otherwise, throw an exception.
+     *
+     * @param  mixed  $key
+     * @param  mixed  $operator
+     * @param  mixed  $value
+     * @return mixed
+     *
+     * @throws \Illuminate\Collections\ItemNotFoundException
+     * @throws \Illuminate\Collections\MultipleItemsFoundException
+     */
+    public function sole($key = null, $operator = null, $value = null)
+    {
+        $filter = func_num_args() > 1
+            ? $this->operatorForWhere(...func_get_args())
+            : $key;
+
+        $items = $this->when($filter)->filter($filter);
+
+        if ($items->isEmpty()) {
+            throw new ItemNotFoundException;
```

Comment:
would be nice if the query builder etc would throw the exact same exceptions when calling sole()

Reply:
https://github.com/laravel/framework/pull/37034#issuecomment-822490782